data/data/bootstrap/baremetal/files/etc/NetworkManager/dispatcher.d/30-local-dns-prepender

```diff
@@ -6,13 +6,26 @@ case "$STATUS" in
     logger -s "NM local-dns-prepender triggered by ${1} ${2}."
     DNS_IP="127.0.0.1"
     set +e
-    logger -s "NM local-dns-prepender: Checking if local DNS IP is the first entry in resolv.conf"
-    if grep nameserver /etc/resolv.conf | head -n 1 | grep -q "$DNS_IP" ; then
-        logger -s "NM local-dns-prepender: local DNS IP already is the first entry in resolv.conf"
-        exit 0
+    if systemctl -q is-enabled systemd-resolved; then
+        >&2 echo "NM resolv-prepender: Setting up systemd-resolved for local DNS"
+        if [[ ! -f /etc/systemd/resolved.conf.d/60-kni.conf ]]; then
+            >&2 echo "NM resolv-prepender: Creating /etc/systemd/resolved.conf.d/60-kni.conf"
+            echo "[Resolve]" > /etc/systemd/resolved.conf.d/60-kni.conf
+            echo "DNS=$DNS_IP" >> /etc/systemd/resolved.conf.d/60-kni.conf
+            if systemctl -q is-active systemd-resolved; then
+                >&2 echo "NM resolv-prepender: restarting systemd-resolved"
+                systemctl restart systemd-resolved
```

Comment:
Hmm too bad it doesn't support `reload` right now, I think this might transiently fail any in-flight DNS lookups (OTOH I guess this is mostly done when the network first comes up).